extensions/incubator/src/main/java/io/opentelemetry/extension/incubator/trace/ExtendedTracer.java

```diff
@@ -5,54 +5,317 @@
+      return spanCallback.doInSpan();
     } catch (Throwable e) {
-      span.recordException(e);
+      handleException.accept(span, e);
       throw e;
     } finally {
       span.end();
     }
   }
 
-  /** Call the provided {@link Callable} and wrap with a {@link Span} with the provided name. */
-  public <T> T call(String spanName, Callable<T> callable) throws Exception {
-    Span span = delegate.spanBuilder(spanName).startSpan();
-    try (Scope scope = span.makeCurrent()) {
-      return callable.call();
-    } catch (Throwable e) {
-      span.recordException(e);
-      throw e;
-    } finally {
-      span.end();
+  /**
+   * Marks a span as error.
+   *
+   * @param span the span
+   * @param exception the exception that caused the error
+   */
+  private static void setSpanError(Span span, Throwable exception) {
+    span.setStatus(StatusCode.ERROR);
+    span.recordException(exception);
+  }
+
+  /**
+   * Injects the current context into a string map, which can then be added to HTTP headers or the
+   * metadata of an event.
+   */
+  public Map<String, String> getTextMapPropagationContext() {
+    return Propagation.getTextMapPropagationContext(openTelemetry);
+  }
+
+  /**
+   * Set baggage items inside the given {@link SpanCallback}.
+   *
+   * @param baggage the baggage items to set
+   * @param spanCallback the {@link SpanCallback} to call
+   * @param <T> the type of the result
+   * @param <E> the type of the exception
+   * @return the result of the {@link SpanCallback}
+   */
+  @SuppressWarnings("NullAway")
+  public static <T, E extends Throwable> T callWithBaggage(
+      Map<String, String> baggage, SpanCallback<T, E> spanCallback) throws E {
+    BaggageBuilder builder = Baggage.current().toBuilder();
+    baggage.forEach(builder::put);
+    Context context = builder.build().storeInContext(Context.current());
+    try (Scope ignore = context.makeCurrent()) {
+      return spanCallback.doInSpan();
     }
   }
 
-  @Override
-  public SpanBuilder spanBuilder(String spanName) {
-    return delegate.spanBuilder(spanName);
+  /**
+   * Run the given {@link SpanCallback} inside a server span.
+   *
+   * <p>The span context will be extracted from the <code>carrier</code>, which you usually get from
+   * HTTP headers of the metadata of an event you're processing.
+   *
+   * <p>If an exception is thrown by the {@link SpanCallback}, the span will be marked as error, and
+   * the exception will be recorded.
+   *
+   * @param carrier the string map where to extract the span context from
+   * @param spanBuilder the {@link SpanBuilder} to use
+   * @param spanCallback the {@link SpanCallback} to call
+   * @param <T> the type of the result
+   * @param <E> the type of the exception
+   * @return the result of the {@link SpanCallback}
+   */
+  public <T, E extends Throwable> T traceServerSpan(
+      Map<String, String> carrier, SpanBuilder spanBuilder, SpanCallback<T, E> spanCallback)
```

Comment:
I'm not particularly fond of the constructors for this class..
- The name ExtendedTracer suggests we might only need a tracer, but we actually need an OpenTelemetry as well
- The `ExtendedTracer(OpenTelemetry openTelemetry, String instrumentationScopeName)` constructor encourages creating a tracer without a version or schemaUrl. 

If these methods were to accept `OpenTelemetry` as an argument (or maybe even `ContextPropagators` or `TextMapPropagators`), then the construction can simplify to being a pure extension of `Tracer`, and only accept a `Tracer` delegate as an argument.

Presumably the goal of methods on ExtendedTracer in the incubator package is to someday have them added to `Tracer`. `Tracer` will never have access to an `OpenTelemetry` instance unless one is provided as an argument, so I think its better to reflect that in the API design of ExtendedTracer as well.

Reply:
> or maybe even `ContextPropagators` or `TextMapPropagators`

👍

Reply:
- provide propagators as params instead of otel instance
- use factory method instead of ctor